curses/client.cpp

```diff
@@ -507,14 +502,14 @@ int main(int argc, char * argv[])
 				{
 					case '+':
 						dac_number++;
-						if (dac_number >= MAX_DACS)
+						if (dac_number >= number_of_dacs)
```

Comment:
This is getting to be a little too deeply nested. Can you pull a few of these scopes into functions?
Also, having two variables called `dac_number` and `number_of_dacs` might get confusing. Is there a way to calculate these on the fly? For example is `number_of_dacs == dac_list.length()` (assuming such list exists)?